src/test/java/ru/curs/celesta/transaction/CelestaTransactionTest.java

```diff
@@ -0,0 +1,85 @@
+package ru.curs.celesta.transaction;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.context.annotation.UserConfigurations;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import ru.curs.celesta.CallContext;
+import ru.curs.celesta.Celesta;
+import ru.curs.celesta.SystemCallContext;
+import ru.curs.celesta.spring.boot.autoconfigure.CelestaAutoConfiguration;
+
+import java.sql.SQLException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CelestaTransactionTest {
+    private static final String SCORE_PATH = "classpath:score";
+
+    private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
+            .withConfiguration(AutoConfigurations.of(CelestaAutoConfiguration.class))
+            .withPropertyValues("celesta.scorePath:" + SCORE_PATH);
+
+
+    @Test
+    void activatesContext() {
+        this.contextRunner
+                .withConfiguration(UserConfigurations.of(DummyService.class))
+                .withPropertyValues("celesta.h2.inMemory:true")
+                .run((context -> {
+                    try (Celesta celesta = context.getBean(Celesta.class)) {
+                        DummyService srv = context.getBean(DummyService.class);
+                        CallContext cc = new CallContext("foo");
+                        assertFalse(cc.isClosed());
+                        assertNull(cc.getCelesta());
+
+                        srv.withContext(cc, "bar");
+
+                        assertNotNull(cc.getCelesta());
+                        assertSame(celesta, cc.getCelesta());
+                        assertTrue(cc.isClosed());
+
+                        shutDownH2(celesta);
+                    }
+                }));
+    }
+
+    @Test
+    void handlesMethodWithNoContext() {
+        this.contextRunner
+                .withConfiguration(UserConfigurations.of(DummyService.class))
+                .withPropertyValues("celesta.h2.inMemory:true")
+                .run((context -> {
+                    try (Celesta celesta = context.getBean(Celesta.class)) {
+                        DummyService srv = context.getBean(DummyService.class);
+                        srv.noContext("test");

```

Comment:
Well, I thought about it. I thought and I thought and then I decided to do this way.

Ideally, it would be great to have a _compile-time_ exception when we use `@CelestaTransaction` annotation on a method without `CallContext` parameter. But runtime exceptions are evil. Imagine someone refactored their service method and got rid of `CallContext` usage but forgot to remove the annotation. It _will_ compile and unit tests _will_ run. For what reason we should throw a runtime exception here? To make software more buggy? The left-over annotation will not produce much problems, since no connection is allocated and no transaction is started when we find no `CallContext` parameter.